File: scripts/update_field.js
const fs = require('fs');
const axios = require('axios');

async function syncFieldWork() {
    const jsonUrl = 'https://rss.app/feeds/v1.1/G9mWxuK3iJIoFhIA.json';
    console.log(`Fetching JSON feed from ${jsonUrl}...`);

    try {
        const response = await axios.get(jsonUrl);
        const data = response.data;

        if (!data || !data.items) {
            console.log("No items found in the feed.");
            return;
        }

        // Keywords to include (Field Work)
        const includeKeywords = [
            'terreno', 'campaña', 'antártica', 'oceanografía', 'muestreo',
            'expedición', 'buque', 'embarque', 'mediciones', 'lanzamiento',
            'instrumento', 'malla', 'estación', 'fiordos', 'patagonia',
            'seals', 'falkor', 'lofisat', 'costar', 'ciencia'
        ];

        // Keywords to exclude (Personal/Family)
        const excludeKeywords = [
            'familia', 'cumpleaños', 'vacaciones', 'niño', 'hijo',
            'asado', 'playa', 'almuerzo', 'descanso', 'personal'
        ];

        const filteredItems = data.items.filter(item => {
            const text = (item.content_text || item.title || '').toLowerCase();

            // Inclusion criteria: must have at least one field keyword
            const hasInclude = includeKeywords.some(kw => text.includes(kw));

            // Exclusion criteria: must not have any personal keyword
            const hasExclude = excludeKeywords.some(kw => text.includes(kw));

            return hasInclude && !hasExclude;
        }).slice(0, 5); // Limit to 5 items to match the layout

        if (filteredItems.length > 0) {
            // Simplified structure for the frontend
            const simplifiedItems = filteredItems.map(item => ({
                id: item.id,
                thumbnail: item.image || (item.attachments && item.attachments[0] ? item.attachments[0].url : ''),
                permalink: item.url,
                caption: item.content_text || item.title || '',
                timestamp: item.date_published
            }));

            fs.writeFileSync('./data/field_posts.json', JSON.stringify(simplifiedItems, null, 2));
            console.log(`Successfully synced ${simplifiedItems.length} field work items.`);
        } else {
            console.log("No items matched the filtering criteria.");
        }

    } catch (error) {
        console.error('Error syncing Field Work JSON:', error.message);
        process.exit(1);
    }
}

syncFieldWork();

File: scripts/update_field.py
import requests
import json
import os
from pathlib import Path

def sync_field_work():
    json_url = 'https://rss.app/feeds/v1.1/G9mWxuK3iJIoFhIA.json'
    print(f"Fetching JSON feed from {json_url}...")
    
    try:
        response = requests.get(json_url, timeout=15)
        if response.status_code != 200:
            print(f"Error: HTTP {response.status_code}")
            return
            
        data = response.json()
        if not data or 'items' not in data:
            print("No items found in the feed.")
            return

        # Keywords to include (Field Work)
        include_keywords = [
            'terreno', 'campaña', 'antártica', 'oceanografía', 'muestreo',
            'expedición', 'buque', 'embarque', 'mediciones', 'lanzamiento',
            'instrumento', 'malla', 'estación', 'fiordos', 'patagonia',
            'seals', 'falkor', 'lofisat', 'costar', 'ciencia'
        ]

        # Keywords to exclude (Personal/Family)
        exclude_keywords = [
            'familia', 'cumpleaños', 'vacaciones', 'niño', 'hijo',
            'asado', 'playa', 'almuerzo', 'descanso', 'personal'
        ]

        filtered_items = []
        for item in data['items']:
            text = (item.get('content_text', '') or item.get('title', '') or '').lower()
            
            has_include = any(kw in text for kw in include_keywords)
            has_exclude = any(kw in text for kw in exclude_keywords)
            
            if has_include and not has_exclude:
                filtered_items.append(item)

        filtered_items = filtered_items[:5] # Limit to 5

        if filtered_items:
            simplified_items = []
            for item in filtered_items:
                thumbnail = item.get('image', '')
                if not thumbnail and item.get('attachments'):
                    thumbnail = item['attachments'][0].get('url', '')
                
                simplified_items.append({
                    "id": item.get('id'),
                    "thumbnail": thumbnail,
                    "permalink": item.get('url'),
                    "caption": item.get('content_text', '') or item.get('title', '') or '',
                    "timestamp": item.get('date_published')
                })

            base_dir = Path(r'd:\Proyectos_atigravity\web_personal')
            output_path = base_dir / 'data' / 'field_posts.json'
            
            with open(output_path, 'w', encoding='utf-8') as f:
                json.dump(simplified_items, f, ensure_ascii=False, indent=2)
            
            print(f"Successfully synced {len(simplified_items)} field work items.")
        else:
            print("No items matched the filtering criteria.")

    except Exception as e:
        print(f"Error syncing Field Work: {e}")

if __name__ == "__main__":
    sync_field_work()
